pingora-core/src/protocols/http/compression/mod.rs

```diff
@@ -571,14 +593,90 @@ fn compressible(resp: &ResponseHeader) -> bool {
     }
 }
 
+// add Vary header with the specified value or extend an existing Vary header value
+fn add_vary_header(resp: &mut ResponseHeader, value: &http::header::HeaderName) {
+    use http::header::{HeaderValue, VARY};
+
+    if let Some(existing) = resp.headers.get(VARY) {
```

Comment:
Sorry for the delayed review. We probably want to use `get_all()` here in case there is more than one `Vary` header on the response?

Reply:
Done